### PR TITLE
Add API level security options

### PR DESCRIPTION
**Pull request by tjholm**
Still need to add operation level security options as well, can be done in a future PR.

### CODE REVIEW

File: resources/api.go [hunk 1]
```diff
@@ -95,21 +98,85 @@ type Api interface {
...
+
+	// Apply options
+	for _, o := range opts {
+		o(a)
+	}
+
+	var secDefs map[string]*v1.ApiSecurityDefinition = nil
+	var security map[string]*v1.ApiScopes = nil
+
+	fmt.Println(a.securityRules)
```
Comment (asalkeld) [inline resources/api.go:129]: do you want this ^ in?

File: resources/api.go [hunk 2]
```diff
@@ -95,21 +98,85 @@ type Api interface {
...
+	securityRules map[string]interface{}
+	security      map[string][]string
 }
 
-func (m *manager) NewApi(name string) Api {
-	return &api{
+func (m *manager) NewApi(name string, opts ...ApiOption) (Api, error) {
+	rsc, err := m.resourceServiceClient()
+
+	if err != nil {
```
Comment (asalkeld) [inline resources/api.go:111]: can we cuddle the error checks please?

```
err := foo()
if err != nil {
    return err
}
```
This is used in all the golang repos I have seen.

**Review by asalkeld — CHANGES_REQUESTED** [review]

File: resources/api.go (hunk 1 above)
Comment (tjholm) [reply resources/api.go]: fixed

File: resources/api.go (hunk 2 above)
Comment (tjholm) [reply resources/api.go:109]: fixed

**Review by asalkeld — APPROVED** [review]

**Pull request merged**